### PR TITLE
chore: update to docusaurus v3.0.0

### LINKED ISSUE #4581: Upgrade to Docusaurus v3
### Is your enhancement related to a problem? Please describe

Docusaurus v3 came out today with some enhancements. Opening an issue because there are a lot of underlying component upgrades so they say migration may be required (i.e. more than a dependabot), and they are already saying v2 will only get security fixes until end of January.

https://docusaurus.io/blog/preparing-your-site-for-docusaurus-v3
https://docusaurus.io/blog/releases/3.0

FYI @themr0c 

### Describe the solution you'd like

Upgrade to Docusaurus v3.

### Describe alternatives you've considered

Docusaurus v2 forever!

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### What does this PR do?
Update website to the new version of docusaurus v3.0.0

https://docusaurus.io/blog/releases/3.0

We should rename all .md files using imports to .mdx but it can be delayed

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/4581

### CODE REVIEW

**Comment by benoitf** [comment]
- icons are broken

**Comment by benoitf** [comment]
icons have been fixed by renaming `<icon` to `<Icon` in https://github.com/containers/podman-desktop/pull/4856

**Comment by benoitf** [comment]
it looks we have block / syntax highlight issues (from screenshots)

**Comment by benoitf** [comment]
ok added the missing `json` and `bash` language for prism (language highlight as they're not enabled by default)

ready to be reviewed

**Comment by benoitf** [comment]
I've rebased this morning due to a conflicting yarn.lock

**Comment by jeffmaury** [comment]
@benoitf can you rebase again

**Comment by benoitf** [comment]
@jeffmaury done

File: packages/main/src/plugin/index.ts [hunk 1]
```diff
@@ -1787,6 +1788,10 @@ export class PluginSystem {
       return kubernetesClient.listPods();
     });
 
+    this.ipcHandle('kubernetes-client:listDeployments', async (): Promise<V1Deployment[]> => {
```
Comment (jeffmaury) [inline packages/main/src/plugin/index.ts]: How is this related to website update ?

Comment (benoitf) [reply packages/main/src/plugin/index.ts]: I have no idea to know what happened

I recreated from a previous working sha

**Comment by benoitf** [comment]
@jeffmaury should be good now

**Review by deboer-tim — APPROVED** [review]
Changes LGTM, compared against current website and could not find differences.

**Comment by benoitf** [comment]
need to rebase again 🤦

**Review by cdrage — APPROVED** [review]
Checked netlify / tested this locally and LGTM!

Let's get this in before we have to rebase / merge a bunch again 👍
